common/dynamicconfig/constants.go

```diff
@@ -2072,6 +2072,11 @@ the number of children greater than or equal to this threshold`,
 		time.Minute*10,
 		`WorkflowTaskRetryMaxInterval is the maximum interval added to a workflow task's startToClose timeout for slowing down retry`,
 	)
+	DiscardSpeculativeWorkflowTaskMaximumEventsCount = NewGlobalIntSetting(
+		"history.discardSpeculativeWorkflowTaskMaximumEventsCount",
+		10,
```

Comment:
I am not sure about this default. Should it be higher?

Reply:
Something to chat with the SDK team about?

PS: Is count the best measure; or should it be size?

Reply:
After some discussions I decided to leave it at 10 for now.